compose/ui/ui-test/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skiko.kt

```diff
@@ -205,30 +228,50 @@ open class SkikoComposeUiTest @InternalTestApi constructor(
     private val testOwner = SkikoTestOwner()
     private val testContext = TestContext(testOwner)
 
-    fun <R> runTest(block: SkikoComposeUiTest.() -> R): R {
-        return composeRootRegistry.withRegistry {
-            withScene {
+    fun runTest(
+        block: suspend SkikoComposeUiTest.() -> Unit
+    ): TestResult {
+        composeRootRegistry.setupRegistry()
+        scene = runOnUiThread(::createUi)
+
+        @OptIn(ExperimentalStdlibApi::class)
+        val testDispatcher = runTestContext[CoroutineDispatcher] as? TestDispatcher
+            ?: StandardTestDispatcher()
+
+        // It's required for a test block to run in a coroutine context with some
+        // elements from the current Recomposer context (e.g. MonotonicFrameClock)
+        val combinedCoroutineContext =
+            scene.compositionContext.effectCoroutineContext
+                .minusKey(CoroutineExceptionHandler.Key)
+                .minusKey(Job.Key)
+                .minusKey(TestCoroutineScheduler.Key)
+                .plus(runTestContext)
+                .plus(testDispatcher)
+
+        return kotlinx.coroutines.test.runTest(
```

Comment:
IDEA seems to think `kotlinx.coroutines.test` isn't necessary here.

Reply:
fixed